internal/core/operations/rooms/remove/executor.go

```diff
@@ -77,7 +77,7 @@ func (e *Executor) Execute(ctx context.Context, op *operation.Operation, definit
 
 	if removeDefinition.Amount > 0 {
 		logger.Info("start removing rooms", zap.Int("amount", removeDefinition.Amount))
-		err := e.removeRoomsByAmount(ctx, op.SchedulerName, removeDefinition.Amount, op)
+		err := e.removeRoomsByAmount(ctx, op.SchedulerName, removeDefinition.Amount, op, removeDefinition.Reason)
```

Comment:
nit: at some point, it will be easier to just pass `removeDefinition` ref to those functions and access whatever field they want 😬